Code/src/main/java/nl/utwente/group10/ui/components/ConnectionAnchor.java

```diff
@@ -34,8 +38,45 @@ public ConnectionAnchor(Block block, CustomUIPane pane) throws IOException {
         fxmlLoader.setController(this);			
 
         fxmlLoader.load();
+        
+        setConnection(null);
     }
 
+    public void setConnection(Connection connection){
+    	this.connection = Optional.ofNullable(connection);
+    }
+    
+    public boolean isConnected(){
+    	return connection.isPresent();
+    }
+    
+    public Optional<Connection> getConnection(){
+    	return connection;
+    }
+    
+    public abstract boolean canConnect();
+    public abstract void disconnect(Connection connection);
+    
+    public Optional<ConnectionAnchor> getOtherAnchor(){
+    	if(isConnected()){
+    		Optional<OutputAnchor> out =getConnection().get().getOutputAnchor();
+    		Optional<InputAnchor> in =getConnection().get().getInputAnchor();
+    		if(in.isPresent() && out.get().equals(this)){
+    			return Optional.of(in.get());
+    		}else if(out.isPresent() && in.get().equals(this)){
+    			return Optional.of(out.get());
+    		}
+    	}
+    	return Optional.empty();
+    }
+    
+    /**
+     * @return The newly made Connection of successful
+     * Note that failing to add a connection while already connected will still return an empty optional.
+     * (No new Connection was created)
```

Comment:
Wellicht handig om dat in de Javadoc te vermelden ook als je dat nog niet gedaan hebt.